lib/db/mem.js

```diff
@@ -1333,23 +1340,25 @@ module.exports = function (log, error) {
           throw error.notFound()
         }
 
-        let foundCode, foundIndex
-        for (let i = 0; i < codes.length; i++) {
-          const code = codes[i]
-          if (codeHash === code.codeHash.toString('hex')) {
-            foundCode = code
-            foundIndex = i
-            break
-          }
-        }
+        const hashes = codes.map((item, index) => {
+          return dbUtil.compareHashScrypt(code, item.codeHash, item.salt)
+            .then((equals) => {
+              return {item, index, equals}
```

Comment:
No biggie but looks like only `index` and `equals` get referenced in the returned object here? It might make the `item` from the block below a little bit clearer if it's there's not an `item` property returned here too?